bundles/org.eclipse.equinox.common/src/org/eclipse/core/runtime/ServiceCaller.java

```diff
@@ -147,7 +150,9 @@ public static <S> boolean callOnce(Class<?> caller, Class<S> serviceType, Consum
 	 * @param <S>         the OSGi service type to look up
 	 * @return true if the OSGi service was located and called successfully, false
 	 *         otherwise
-	 * @throws NullPointerException if any of the parameters are null
+	 * @throws NullPointerException  if any of the parameters are {@code null}
+	 * @throws IllegalStateException if {@link FrameworkUtil#getBundle(Class)}
```

Comment:
This exposes implementation details that could change in the future.  Instead can we have something like:

> if the bundle associated with the caller class cannot be determined